### ISSUE TITLE
Forwarding scoped slot does not map to the correct slot assignment

### ISSUE THREAD

**Issue by jmsjtu** [salesforce/lwc#3889]
### Description

When forwarding a scoped slot to another slot, the slot mapping is incorrect. The scoped slot will always be slotted into the default slot of the component.

Here's a stackblitz [demo](https://stackblitz.com/edit/salesforce-lwc-eqjnb1?file=src%2Fmodules%2Fx%2Fapp%2Fapp.html).

```html
<template lwc:render-mode="light">
  <x-scoped-slot-child>
    <template lwc:slot-data="title">
      <h2>{title}</h2>
    </template>
  </x-scoped-slot-child>
</template>

// x-scoped-slot-parent
```

```html
<template lwc:render-mode="light">
  <x-leaf>
    <!-- Note slot="foo" doesn't map to <slot name="foo"> in x-leaf -->
    <slot lwc:slot-bind={title} slot="foo"></slot>
  </x-leaf>
</template>

// x-scoped-slot-child
```
Notice in `x-scoped-slot-child`, `slot="foo"` on the `slot` element.

```html
<template>
  <slot></slot>
  <slot name="foo"></slot>
</template>

// x-leaf
```

Inspecting the elements, the scoped slot is slotted into the default slot of `x-leaf`, ignoring the `slot` attribute in `x-scoped-slot-child`.

**Comment by jmsjtu** [comment]
cc @ravijayaramappa created an issue to track

**Comment by jmsjtu** [comment]
Fix with #3883
